torch/onnx/_internal/fx/passes/readability.py

```diff
@@ -0,0 +1,111 @@
+        for node in nodes:
+            with self.module.graph.inserting_before(node):
+                new_node = self.module.graph.get_attr(normalized_name)
+                new_node.meta = node.meta
+                node.replace_all_uses_with(new_node)
+                self.module.graph.erase_node(node)
+        diagnostic.with_additional_message(
+            f"Renamed 'self.{old_name}' to 'self.{normalized_name}', "
+            f"normalized from original parameter name '{new_name}'."
+        )
+
+    def _run(self, *args, **kwargs) -> torch.fx.GraphModule:
+        """Restore parameter and buffer names from original module.
+
+        For each `get_attr` node, if the target is a str representing a parameter or buffer
+        under `self.module`, we rename the parameter or buffer to its original name.
+        The parameters and buffers between `self.module` and `self.original_module` refer
+        to the same objects, allowing us to use it as key to retrieve the original name.
+        """
+        assert len(args) == 0, "RestoreParameterAndBufferNames does not take any args"
+        assert (
+            len(kwargs) == 0
+        ), "RestoreParameterAndBufferNames does not take any kwargs"
+        state_to_readable_name: Dict[Union[torch.nn.Parameter, torch.Tensor], str] = {}
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_parameters()}
+        )
+        state_to_readable_name.update(
+            {v: k for k, v in self.original_module.named_buffers()}
+        )
+        diagnostic = self.diagnostic_context.inflight_diagnostic()
+
+        old_name_to_nodes: Dict[str, Tuple[List[torch.fx.Node], str]] = {}
+
+        for node in self.module.graph.nodes:
+            if node.op == "get_attr":
+                assert isinstance(
+                    node.target, str
+                ), f"Expected str, got type({node.target})"
+                if node.target in old_name_to_nodes:
+                    # We have already processed this parameter/buffer.
+                    old_name_to_nodes[node.target][0].append(node)
+                    continue
+                attr_value = getattr(self.module, node.target)
+                if (
+                    isinstance(attr_value, (torch.nn.Parameter, torch.Tensor))
+                    and attr_value in state_to_readable_name
+                ):
+                    readable_name = state_to_readable_name[attr_value]
+                    old_name_to_nodes[node.target] = ([node], readable_name)
+                    continue
+
+                diagnostic.with_additional_message(
+                    f"Cannot find readable name for self.{node.target}: {type(attr_value)}. The name is unchanged."
+                )
+                if isinstance(attr_value, torch.nn.Parameter):
+                    # If it is a parameter we treat it more seriously.
+                    diagnostic.level = diagnostics.levels.WARNING
```

Comment:
Alter state is not recommended. Maybe keep it untouched?

Reply:
We are logging content into the diagnostic object throughout the execution of the function. The intent here is to increase the severity of a diagnostic when certain issues were found. Do you have suggestions for a better api?

Reply:
`diagnistic.with_local_level(WARNING)`